ADAL/src/ADAuthenticationErrorConverter.m

```diff
@@ -69,13 +72,12 @@ + (ADAuthenticationError *)ADAuthenticationErrorFromMSIDError:(NSError *)msidErr
         domain = s_errorDomainMapping[domain];
     }
     
-    //Map errorCode. Note that errorCode must be mapped together with domain
+    //Map errorCode
     NSInteger errorCode = msidError.code;
-    NSString *mapKey = [NSString stringWithFormat:@"%@|%ld", msidError.domain, (long)errorCode];
-    NSString *mapValue = s_errorCodeMapping[mapKey];
-    if (![NSString msidIsStringNilOrBlank:mapValue])
+    NSNumber *mappedErrorCode = s_errorCodeMapping[msidError.domain][NSNUMBER(msidError.code)];
+    if (mappedErrorCode)
     {
-        errorCode = [mapValue integerValue];
+        errorCode = [mappedErrorCode integerValue];
```

Comment:
I'm not sure we should continue if no mapped error code found.

Reply:
There are two mapping cases here:
1. MSIDSomeErrorDomain + MSIDSomeErrorCode being mapped to ADSomeErrorDomain + ADSomeErrorCode
2. MSIDSomeErrorDomain being mapped to ADSomeErrorDomain but without changing the error code. 
E.g., for `ADHTTPErrorCodeDomain`, we just want to map the error domain from `MSIDHTTPErrorCodeDomain` to `ADHTTPErrorCodeDomain` but keep the error code unchanged. Therefore we continue without changing the error code for case 2.

Reply:
Ok, I see :) Can we differentiate between cases when there's no mapping found because we forgot to add mapping and expected ones? e.g. for ADErrors we should always have a mapping.